### ISSUE TITLE
why the nested `pluginProps`?

### ISSUE THREAD

**Issue by ianstormtaylor** [draft-js-plugins/draft-js-plugins#143]
Curious to know what the reason is for the `pluginProps` object being nested, instead of just returning a `plugin` object that the `Editor` knows how to use directly? If the nesting wasn't there it would make it easy to write private plugins inside one's own repo as:

```js
export function handleKeyCommand(command) {
  ...
}

export function keyBindingFn(e) {
   ...
}
```

**Comment by nikgraf** [comment]
We had that in beginning. I noticed people got a bit confused about which part is custom and which are function hooks used by the plugin system. Moving them to pluginProps helped. I don't have hard feelings on reverting that change to the old structure. Good documentation could solve the confusion part.

Thoughts? @ianstormtaylor @juliankrispel @bkniffler @mxstbr

**Comment by bkniffler** [comment]
I think that separating editor/plugin-props makes sense logically. Mixing them with custom properties, hooks, whatever a plugin can define thats not an editor prop wouldn't be a clean thing to do, but it wouldn't do any harm neither as naming conflicts are quite unlikely. And as most plugins will only define pluginProps, it might be beneficial to get rid of them.

**Comment by juliankrispel** [comment]
I came to the same conclusion as @nikgraf tbh.

Having a flat level object also wouldn't work because you wouldn't be able to do multiple strategy/component pairs...

**Comment by juliankrispel** [comment]
I'll close this issue for now, we can still discuss on or reopen it if we change our mind...

**Comment by ianstormtaylor** [comment]
Why are they named `pluginProps` if they map to the editor's hooks? Why not something terser/cleaner at least like `hooks`? 

@juliankrispel not sure what you mean by multiple strategy/component pairs, wouldn't that be handled already because they are nested inside `.decorators`?

**Comment by nikgraf** [comment]
@ianstormtaylor yes, you are right about `decorators`

re pluginsProps: there are not hooks, decorators are rather an object with the strategy & component

Right now I'm thinking to going back to a flat structure as well. Let me sleep over this one more night.

**Comment by juliankrispel** [comment]
Right, let's sum this up. Here's what the config looks like right now.

```js
{
  pluginProps: {
    decorators: [ {strategy, component} ],
    getEditorProps: () => {},
    ...eventHandlers
  }
}
```

I looked into this a tad more and I more or less agree with @ianstormtaylor. The configuration is called pluginProps because right now that configuration is actually mounted directly on the Editor. I think that is bad and I created a separate issue for it: #147

In light of this matter, how about 

A) A flat object like @ianstormtaylor suggested
or 
B) `{ decorators: [...], eventHandlers: [...] }` to separate the two more explicitly
